tcti-device: change device tcti to nonblocking mode

tstruk opened the pull request:
Change device tcti to nonblocking mode of operation.

flihp commented on configure.ac:
```diff
@@ -64,13 +64,22 @@ AC_ARG_ENABLE([esapi],
             [enable_esapi=$enableval],
             [enable_esapi=yes])
 
-
```
whitespace. This is good cleanup but not related to this PR / feature.

flihp commented on configure.ac:
```diff
@@ -64,13 +64,22 @@ AC_ARG_ENABLE([esapi],
+		    (note: This needs to be supported by the kernel driver). default is no]),
+    [enable_tcti_device_async=$enableval],
+    [enable_tcti_device_async=no])
+AM_CONDITIONAL(TCTI_ASYNC, test "x$enable_tcti_device_async" = "xyes")
```
I don't see a place in the makefile where you use this variable. Unless I missed it, this isn't necessary.

tstruk replied:
Removed

flihp commented on src/tss2-tcti/tcti-device.c:
```diff
@@ -118,7 +119,7 @@ tcti_device_receive (
     TSS2_TCTI_COMMON_CONTEXT *tcti_common = tcti_device_down_cast (tcti_dev);
     TSS2_RC rc = TSS2_RC_SUCCESS;
-    ssize_t  size;
+    ssize_t size = 0;
```
This good cleanup but it doesn't belong in this commit.

tstruk replied:
The size needs to be initialized because the compiler complains otherwise

flihp commented on src/tss2-tcti/tcti-device.c:
```diff
@@ -251,6 +294,7 @@ Tss2_Tcti_Device_Init (
     TSS2_TCTI_DEVICE_CONTEXT *tcti_dev;
     TSS2_TCTI_COMMON_CONTEXT *tcti_common;
     const char *dev_path = conf != NULL ? conf : TCTI_DEVICE_DEFAULT;
+    mode_t mode = O_RDWR;
```
This isn't a mode, it's a flag since it's used in the second parameter: `int open(const char *pathname, int flags, mode_t mode);`. The reason you're not seeing a type error is that the `flags` paramter is an `int` and `mode_t` maps back to `int`. So this variable should be an `int` and it should be called `flags`.

tstruk replied:
Fixed

tstruk commented:
@flihp Thanks for your feedback. I have reworked the code accordingly, updated the commit message, and added unit tests to cover the poll() interface. Please have a look.
